airflow/models.py

```diff
@@ -867,6 +873,7 @@ def __init__(self, task, execution_date, state=None):
         self.priority_weight = task.priority_weight_total
         self.try_number = 0
         self.max_tries = self.task.retries
+        self.job_id = job_id
```

Comment:
Rather: you add a param to the constructor but don't seem to use it (and the pattern the rest of the code uses is just to set `ti.job_id = job_id`

Reply:
Okay, I saw the previous comments on this. Rather than adding a new form to the constructor we should just set the property on the instance in the tests (around line 1363)

Reply:
@ashb thanks for the suggestion. Changed the spec code, so no changes required in here.